controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -368,33 +392,39 @@ public void listStreamsInScope(Controller.StreamsInScopeRequest request, StreamO
         String scope = request.getScope().getScope();
         RequestTag requestTag = requestTracker.initializeAndTrackRequestTag(requestIdGenerator.get(), "listStream", scope);
         log.info(requestTag.getRequestId(), "listStream called for scope {}.", scope);
-        authenticateExecuteAndProcessResults(() -> this.authHelper.checkAuthorization(scope, AuthHandler.Permissions.READ),
-                delegationToken -> controllerService.listStreams(scope, request.getContinuationToken().getToken(), listStreamsInScopeLimit)
-                                                    .thenApply(response -> {
-                                                        List<StreamInfo> streams = response.getKey().stream()
-                                                                                    .map(m -> StreamInfo.newBuilder().setScope(scope).setStream(m).build())
-                                                                                    .collect(Collectors.toList());
-                                                        return Controller.StreamsInScopeResponse.newBuilder().addAllStreams(
-                                                                streams).setContinuationToken(Controller.ContinuationToken
-                                                                .newBuilder().setToken(response.getValue()).build()).build();
-                                                    }),
-                responseObserver, requestTag);
+        authenticateExecuteAndProcessResults(
+                () -> this.authHelper.checkAuthorization(AuthResourceRepresentation.ofAScope(scope),
+                        AuthHandler.Permissions.READ),
+                delegationToken -> controllerService.listStreams(
+                        scope, request.getContinuationToken().getToken(), listStreamsInScopeLimit)
+                        .thenApply(response -> {
+                             List<StreamInfo> streams = response.getKey().stream()
+                                     .filter(m -> authHelper.isAuthorized(m, AuthHandler.Permissions.READ))
```

Comment:
This is one of the semantic changes, it filters the elements of the list according to visibility, right?

Reply:
Yes, that's true.